pkg/controllers/cluster/taint_manager.go

```diff
@@ -267,9 +263,9 @@ func (tc *NoExecuteTaintManager) syncClusterBindingEviction(key util.QueueKey) e
 	return nil
 }
 
-// needEviction returns whether the binding should be evicted from target cluster right now.
-// If a toleration time is found, we return false along with a minimum toleration time as the
-// second return value.
+// needEviction returns whether the binding should be evicted from the target cluster right now.
+// If it returns false, further check of the toleration time is needed. A toleration time less than 0
+// means tolerate forever; otherwise, eviction need happen after the toleration time.
```

Comment:
> If it returns false, further check of the toleration time is needed.

This confuses me as not clear who should `further check`, it can be this function itself or the caller outside.

```suggestion
// needEviction determines if a binding should be evicted from the specified cluster based on taints and tolerations.
// Returns:
//   - needEviction: true if the binding should be evicted (when to evict is indicated by the second return value).
//   - tolerationTime: if needEviction is true, this is the duration to wait before eviction (0 means evict immediately,
//     >0 means evict after this duration).
//     If needEviction is false, a value < 0 indicates the binding should be tolerated indefinitely (no eviction).
//   - error: any error encountered during evaluation.
```

Reply:
Actually, this function doesn't have to return `toleration time : -1`, as if the first return value shows if a binding needs to be evicted. Right?